### ISSUE TITLE
Use module aliases instead of packs.

### ISSUE THREAD

**Issue by jordwalke** [ocaml-obuild/obuild#131]
It's known that module aliases increase build speed because they don't need to re-include the entire module implementation.

My project CommonML, uses module aliases to accomplish namespacing and I'd love to see `obuild` do the same:

https://github.com/jordwalke/commonml

**Comment by UnixJunkie** [comment]
What do you mean?
Do you want the code of obuild to be changed because obuild doesn't build fast enough?

**Comment by jeromemaloberti** [comment]
No, obuild makes packs for libraries, with recent versions of ocaml, it could use module aliases, which is much faster at link time, afaik.
It is a good idea, I guess I just need to figure out from which version of ocaml module aliases are available, and what is the cli invocation.

**Comment by UnixJunkie** [comment]
I see from version 4.02.0 by Alain Frisch in this file:
http://caml.inria.fr/pub/distrib/ocaml-4.02/notes/Changes

**Comment by UnixJunkie** [comment]
https://blogs.janestreet.com/better-namespaces-through-module-aliases/
also confirms 4.02.0

**Comment by jordwalke** [comment]
You can see how I do it in CommonML - just look at the log output when it compiles the examples.
https://github.com/jordwalke/commonml

Basically, if you invoke the right compilation commands, you get the equivalent of pack namespacing according to file name.
